docs/src/main/sphinx/connector/kafka.rst

```diff
@@ -653,19 +652,21 @@ Example insert query for the above table definition::
     INSERT INTO example_raw_table (field1, field2, field3, field4)
       VALUES (123456789, 123456, 1234, 'abcdef');
 
-When inserting variable width types, the value must be exactly equal to the
-width defined in the table definition file. If the inserted value is longer it
-gets truncated, resulting in data loss. If the inserted value is shorter the
-decoder will not be able to properly read the value because there is no defined
-padding character. Due to these constraints the encoder fails if the width of
-the inserted value is not equal to the mapping width.
+.. note::
+
+    When inserting variable width types, the value must be exactly equal to the
+    width defined in the table definition file. If the inserted value is longer
+    it gets truncated, resulting in data loss. If the inserted value is shorter,
```

Comment:
This suggests we can insert values longer than the width but leads to data loss. Last sentence in this note says otherwise that the encoder fails in such case preventing a correctness issue - this is a note to myself to find the true behaviour and clarify this statement.

Reply:
I now understand what this note is trying to convey. The first two sentences provide justification for why we don't allow inserting shorter or longer values when using the `RAW` encoder. The final sentence tells what the actual behaviour is.

Can we improve this like below? WDYT?

```
The raw encoder requires the field size to be known ahead of time including for variable width data types like `VARCHAR` and disallows inserting values which don't exactly match the width defined in the table definition file. This is done to ensure correctness otherwise longer values would get truncated and shorter values won't be read back correctly due to lack of a defined padding character.
```

Reply:
If needed we can also mention what the error looks like in this case:

```
length '6' of message 'ashhar' for column 'c_varchar' does not equal expected length '5'
```

(this is with a field like below)

```json
      {
        "name": "c_varchar",
        "type": "VARCHAR",
        "mapping": "0:5"
      },
```

Reply:
We don't generally include error messages. Will leave that out.